### PR TITLE
Limit batch size in delete and change visibility

### LINKED ISSUE #8478: bug: max SQS batch size must be always 10
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I can delete messages with batch sizes greater than 10 without any errors.

The same applies to changing message visibility.

### Expected Behavior

SQS allows processing batches with a maximum size of 10 elements.

### How are you starting LocalStack?

With a `docker run` command

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

    docker run --rm -it -p 4566:4566 -p 4510-4559:4510-4559 localstack/localstack

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

```python
import boto3

sqs = boto3.client("sqs", endpoint_url="http://localhost:4566")
queue_url = sqs.create_queue(QueueName="my-queue")["QueueUrl"]

sqs.send_message_batch(
    QueueUrl=queue_url,
    Entries=[{"Id": str(i), "MessageBody": "foo"} for i in range(10)],
)
sqs.send_message_batch(
    QueueUrl=queue_url,
    Entries=[{"Id": str(i), "MessageBody": "foo"} for i in range(10)],
)

messages = []
messages.extend(
    sqs.receive_message(
        QueueUrl=queue_url,
        AttributeNames=["All"],
        MaxNumberOfMessages=10,
    )["Messages"]
)
messages.extend(
    sqs.receive_message(
        QueueUrl=queue_url,
        AttributeNames=["All"],
        MaxNumberOfMessages=10,
    )["Messages"]
)

# This should raise an error because we're providing a batch with more than 10 elements
sqs.change_message_visibility_batch(
    QueueUrl=queue_url,
    Entries=[
        {
            "Id": str(i),
            "ReceiptHandle": msg["ReceiptHandle"],
            "VisibilityTimeout": 123,
        }
        for i, msg in enumerate(messages)
    ],
)

# This should raise an error because we're providing a batch of more than 10 elements
sqs.delete_message_batch(
    QueueUrl=queue_url,
    Entries=[
        {"Id": str(i), "ReceiptHandle": msg["ReceiptHandle"]}
        for i, msg in enumerate(messages)
    ],
)
```

### Environment

```markdown
- OS: Ubuntu 22.04
- LocalStack: latest
```

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by sinecode**
Fix #8478

### CODE REVIEW

**Review by baermat — CHANGES_REQUESTED** [review]
Thank you @sinecode for your contribution, and thank you for the detailed bug report! This already looks very good, I only have a couple of smaller things that need to be addressed, then we can merge this!

**Comment by baermat** [comment]
I also ran the tests (with the hashtag removed and the PR rebased against master) and it looks all good :+1:

File: localstack/services/sqs/provider.py [hunk 1]
```diff
@@ -1199,6 +1194,10 @@ def _validate_actions(self, actions: ActionNameList):
     def _assert_batch(self, batch: List) -> None:
         if not batch:
             raise EmptyBatchRequest
+        if batch and (no_entries := len(batch)) > MAX_NUMBER_OF_MESSAGES:
+            raise TooManyEntriesInBatchRequest(
+                f"Maximum number of entries per request are #{MAX_NUMBER_OF_MESSAGES}. You have sent {no_entries}."
+            )
```
[the remark this answers is not in the document]
Comment (sinecode) [reply localstack/services/sqs/provider.py]: Sorry, that's muscle memory from Ruby 😄

File: tests/integration/test_sqs.py [hunk 2]
```diff
@@ -1946,6 +1946,102 @@ def test_delete_message_batch_invalid_msg_id(
...
+        assert len(successful) == len(message_batch)
+
+        result_recv = []
+        while len(result_recv) <= MAX_NUMBER_OF_MESSAGES:
+            result_recv.extend(
+                aws_client.sqs.receive_message(
+                    QueueUrl=queue_url, MaxNumberOfMessages=MAX_NUMBER_OF_MESSAGES
+                )["Messages"]
+            )
+        assert len(result_recv) > MAX_NUMBER_OF_MESSAGES
```
[the remark this answers is not in the document]
Comment (sinecode) [reply tests/integration/test_sqs.py]: Didn't know about the `retry` function! I just made a small change to be sure to get the right quantity of messages at each iteration:

```
        result_recv = []
        target_size = 2 * MAX_NUMBER_OF_MESSAGES

        def _receive_all_messages():
            result_recv.extend(
                aws_client.sqs.receive_message(
                    QueueUrl=queue_url,
                    MaxNumberOfMessages=min(MAX_NUMBER_OF_MESSAGES, target_size - len(result_recv)),
                )["Messages"]
            )
            assert len(result_recv) == target_size

        retry(_receive_all_messages, retries=7, sleep=0.5)
```

**Review by baermat — APPROVED** [review]
Sorry for the delay, discussed this and it looks good now :+1:
Thank you again for your contribution :slightly_smiling_face:

**Pull request merged**